service/awsconfig/v5/resource/cloudformation/current.go

```diff
@@ -16,103 +18,129 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return StackState{}, microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "debug", "looking for AWS stack")
-
+	// In order to compute the current state of the guest cluster's main stack we
+	// have to describe the CF stacks and lookup the right stack. We dispatch our
+	// custom StackState structure and enrich it with all information necessary to
+	// reconcile the cloudformation resource.
 	stackName := key.MainGuestStackName(customObject)
+	describeOutput := &cloudformation.DescribeStacksOutput{}
+	{
+		r.logger.LogCtx(ctx, "debug", "looking for main stack")
 
-	describeInput := &cloudformation.DescribeStacksInput{
-		StackName: aws.String(stackName),
-	}
-	describeOutput, err := r.clients.CloudFormation.DescribeStacks(describeInput)
-	if IsStackNotFound(err) {
-		r.logger.LogCtx(ctx, "debug", "did not find a stack in AWS API")
-		return StackState{}, nil
-	} else if err != nil {
-		return StackState{}, microerror.Mask(err)
-	}
+		describeInput := &cloudformation.DescribeStacksInput{
+			StackName: aws.String(stackName),
+		}
+		describeOutput, err = r.clients.CloudFormation.DescribeStacks(describeInput)
+		if IsStackNotFound(err) {
+			r.logger.LogCtx(ctx, "debug", "did not find main stack")
+			return StackState{}, nil
+		} else if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		if len(describeOutput.Stacks) > 1 {
+			return StackState{}, microerror.Mask(notFoundError)
+		}
 
-	if len(describeOutput.Stacks) > 1 {
-		return StackState{}, microerror.Mask(notFoundError)
+		r.logger.LogCtx(ctx, "debug", "found main stack")
 	}
 
 	// GetCurrentState is called on cluster deletion, if the stack creation failed
 	// the outputs can be unaccessible, this can lead to a stack that cannot be
 	// deleted. it can also be called during creation, while the outputs are still
 	// not accessible.
-	status := describeOutput.Stacks[0].StackStatus
-	errorStatuses := []string{
-		"ROLLBACK_IN_PROGRESS",
-		"ROLLBACK_COMPLETE",
-		"CREATE_IN_PROGRESS",
-	}
-	for _, errorStatus := range errorStatuses {
-		if *status == errorStatus {
-			outputStackState := StackState{
-				Name: stackName,
+	status := *describeOutput.Stacks[0].StackStatus
+	{
+		errorStatuses := []string{
+			"ROLLBACK_IN_PROGRESS",
+			"ROLLBACK_COMPLETE",
+			"CREATE_IN_PROGRESS",
+		}
+
+		for _, errorStatus := range errorStatuses {
+			if status == errorStatus {
+				outputStackState := StackState{
+					Name: stackName,
+				}
+
+				return outputStackState, nil
 			}
-			return outputStackState, nil
 		}
 	}
 
-	outputs := describeOutput.Stacks[0].Outputs
+	// In case the current guest cluster is already being updated, we cancel the
+	// reconciliation until the current update is done in order to reduce
+	// unnecessary friction.
+	if status == cloudformation.ResourceStatusUpdateInProgress {
```

Comment:
This here is the cancelation I added.

Reply:
Note this cancels the reconciliation for the all resources. We also have the option to only cancel for only the cloudformation resource. What do you think makes more sense? For instance the S3 object resources run before this already.

Reply:
The cloudformation resource is intentionally the last AWS resource. So only canceling it makes sense to me.

Note: The CF resource is last because we can't create S3 objects using CF. So the KMS and S3 resources need to run first.

Same applies for loadbalancer and ebsvolume because they need to delete resources before the CF resource deletes the stacks.

Reply:
So, `namespaceResource`, `serviceResource` and `endpointsResource` won't be reconciled? Not sure if something needs to be updated on any of those during CF update, maybe better stay in the safe side and only cancel CF?

Reply:
Those resources are only for the guest cluster scraping. If the private IP of the master has changed the endpoints resource needs to be reconciled.

Maybe it is safer to cancel all remaining resources. Not a strong opinion that we just cancel the CF resource.

Reply:
That is all not written in stone. We can just see how it goes with canceling only the cloudformation resource and adapt if necessary.

Reply:
Done.